api/src/main/java/edu/cornell/mannlib/vitro/webapp/auth/objects/AccessObject.java

```diff
@@ -0,0 +1,106 @@
+/* $This file is distributed under the terms of the license in LICENSE$ */
+
+package edu.cornell.mannlib.vitro.webapp.auth.objects;
+
+import edu.cornell.mannlib.vitro.webapp.auth.attributes.AccessObjectType;
+import edu.cornell.mannlib.vitro.webapp.beans.DataProperty;
+import edu.cornell.mannlib.vitro.webapp.beans.ObjectProperty;
+import edu.cornell.mannlib.vitro.webapp.beans.Property;
+import org.apache.jena.rdf.model.Model;
+
+public abstract class AccessObject {
+
+    public static String SOME_URI = "?SOME_URI";
+    public static Property SOME_PREDICATE = new Property(SOME_URI);
+    public static String SOME_LITERAL = "?SOME_LITERAL";
+
+    protected AccessObjectStatement statement;
+    private DataProperty dataProperty;
+    private ObjectProperty objectProperty;
+
+    public ObjectProperty getObjectProperty() {
+        return objectProperty;
+    }
+
+    public void setObjectProperty(ObjectProperty objectProperty) {
+        this.objectProperty = objectProperty;
+    }
+
+    public String getUri() {
+        return null;
+    }
+
+    public abstract AccessObjectType getType();
+
+    public AccessObjectStatement getStatement() {
+        return statement;
+    };
+
+    protected void initializeStatement() {
+        if (statement == null) {
+            statement = new AccessObjectStatement();
+        }
+    }
+
+    public void setStatementOntModel(Model ontModel) {
+        initializeStatement();
+        getStatement().setModel(ontModel);
+    }
+
+    public Model getStatementOntModel() {
+        initializeStatement();
+        return getStatement().getModel();
+    }
+
+    public void setStatementSubject(String subject) {
+        initializeStatement();
+        getStatement().setSubject(subject);
+    }
+
+    public String getStatementSubject() {
+        initializeStatement();
+        return getStatement().getSubject();
+    }
+
+    public void setStatementPredicate(Property predicate) {
+        initializeStatement();
+        getStatement().setPredicate(predicate);
+    }
+
+    protected Property getPredicate() {
+        initializeStatement();
+        return getStatement().getPredicate();
+    }
+
+    public String getStatementPredicateUri() {
+        if (statement == null || statement.getPredicate() == null) {
+            return null;
+        }
+        Property predicate = getPredicate();
+        return predicate.getURI();
+    }
```

Comment:
Why the exception here? Why not AccessObjectStatement() as it is used for getPredicate(), etc?

Reply:
I think there is no need to initialize statement and get predicate, get uri of null predicate if statement not exists or predicate is null.

Reply:
ok, let's keep this as is now